openapi/api.yml

```diff
@@ -210,6 +230,29 @@ components:
               type: object
               description: A map of the highest role for each service
               example: {"COF": "ASSESSOR", "NSTF": "COMMENTER"}
+    accountSearch:
+      type: object
+      nullable: true
+      additionalProperties: false
+      properties:
+        email_domain:
+          description: "Search by email domain"
+          example: "communities.gov.uk"
+          type: string
+        roles:
+          description: "Filter results to accounts that have ANY of these roles."
+          example: ['SECTION_151', 'COF_ASSESSOR_R1']
+          type: array
+          items:
+            type: string
+        partial_roles:
+          description: "Filter results to accounts that have roles partially matching ANY of these values."
+          example: ['ASSESSOR']
+          type: array
+          items:
+            type: string
+      not:
```

Comment:
Again, asking questions here for my own education - here we're saying `roles` and `partial_roles` aren't required, but we haven't said that `email_domain` is required so technically they're all optional, is that right? Is there a reason we're doing it this way rather than saying eg. `email_domain` is required?

Reply:
All three fields are optional.

The funkiness here is saying that we don't accept any requests that provide _both_ `roles` and `partial_roles`. It doesn't read super clearly using the wording here. `not` followed by `required` says "these things cannot all be present", rather than "these two things are not required" 🙃

Reply:
Gotcha! Thanks!